lib/dao.js

```diff
@@ -1169,6 +1170,14 @@ DataAccessObject._coerce = function (where) {
                 throw err;
               }
               break;
+            case 'regexp':
+              try {
+                val = utils.toRegExp(val);
+              } catch (err) {
+                err.statusCode = 400;
+                throw err;
```

Comment:
Yes we can return err and have the caller check return values for instance of Error before proceeding instead if you prefer.